share/wake/lib/system/remote_cache_api.wake

```diff
@@ -325,6 +334,27 @@ export def rscApiFindMatchingJob (req: CacheSearchRequest) (api: RemoteCacheApi)
 
     mkCacheSearchResponse json
 
+# rscApiCheckAuthorization: Checks if the provided authorization key is valid. 
+#
+# ```
+#  api | rscApiCheckAuthorization = Pass Unit
+# ```
+export def rscApiCheckAuthorization (api: RemoteCacheApi): Result Unit Error =
+    require Some auth = api.getRemoteCacheApiAuthorization
+    else failWithError "rsc: Checking authorization requires authorization but none was provided"
+
+    require Pass response =
```

Comment:
How frequently is this function called?
The db should cache the HTTPRequest job but I'm curious if the wake function should be a target?

Reply:
> How frequently is this function called?

Two answers here
1. As much as the consuming client calls it
2. In the case of RemoteCacheRunner/default wake implementation, only once per wake invocation so a target doesn't change anything here

Aside: 
> The db should cache the HTTPRequest job

This isn't the case. Web requests are never safe to cache as the result can change at any time. Because of that web request jobs are never stored in the db

Reply:
SGTM.